### PR TITLE
Historical view v1

### PR DESCRIPTION
**Pull request by Kongkille**
First version of historical view addressing #3938.

This PR introduces 4 different time ranges that users can use to explore past data. This also contains large infrastructure changes including large changes to our redux setup as well as changes to the zonemap allowing it to change on every hour.

## Preview

image

### CODE REVIEW

**Review by VIKTORVAV99 — COMMENTED** [review]
Just took a quick look at the code changes and didn't find anything major but I left 2 suggestions for the ci (workflow) files.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -75,6 +75,12 @@ jobs:
     if: needs.setup.outputs.markdown != 'false' || needs.setup.outputs.workflows != 'false'
     uses: ./.github/workflows/validate-local-links-in-md.yml
 
+  jest:
+    name: Jest
+    needs: [setup]
+    if: needs.setup.outputs.frontend != 'false'
```
[the remark this answers is not in the document]
Comment (Kongkille) [reply .github/workflows/CI.yml:81]: Make sense!

**Comment by Kongkille** [comment]
> Just took a quick look at the code changes and didn't find anything major but I left 2 suggestions for the ci (workflow) files.

Thanks! Most of this code should also have been reviewed throughout so hopefully, it all works.

In either case, we'll probably run into a few issues since the infrastructure and UI has been changed so drastically but i'll be monitoring a lot over the next couple of weeks :-)

**Comment by Kongkille** [comment]
> @Kongkille let me know if you need a Approving review, I haven't had time to test it all locally yet but like you said before it should already have passed several reviews.

All good merging now. There are some internal deploy processes i need to run but hopefully it's up in a few hours! :)

**Comment by VIKTORVAV99** [comment]
> > @Kongkille let me know if you need a Approving review, I haven't had time to test it all locally yet but like you said before it should already have passed several reviews.
> 
> All good merging now. There are some internal deploy processes i need to run but hopefully it's up in a few hours! :)

Great! Looking forward to seeing it all in action! 🎉

**Pull request merged**